### DISCUSSION TITLE
Organising forms in Tabs for overview and easy navigation

### DISCUSSION THREAD

**Discussion by freddepettersson** [tellurianinteractive/Tellurian.Trains.ModulesRegistryApp#19] [Ideas]
Hi
I tested a lot in YP during the years and also made the translation to swedish .
In the new register , i found Goods function.
If goods are a general function to add goods types, then I understand , and the boxes are för translations , correct?

My suggestion are to add a YP page in the Module editor with TABS in the top of the page, suggestion of tabs
Module | Goods | etc 

/Fredrik

**Comment by fjallemark** [comment]
I have thought of having tabs. The thing is how tabs should be handled; you save each tab or you edit all tabs and then save. It will be very complex to have a global save for all tabs, but quite easy to add tabs if each tab form is saved separately.

**Reply by freddepettersson** [reply]
Hi
Separate save on each tab are ok , I think.
The standard question if you leave the page, " Do you want to save before leaving" ?

**Reply by fjallemark** [reply]
Adding "Do you want to save before leaving?" must stop navigation if the form data is modified but not saved.
The feature to intercept and stop you from navigation will be added later this year.

**Comment by fjallemark** [comment]
Tabs navigation requires checking for changed data and prevent navigation. This is not available in Blazor and is discussed in this issue: https://github.com/dotnet/aspnetcore/issues/14962. When it becomes available, implementing Tabs will be considered.

**Comment by fjallemark** [comment]
This [article](https://shauncurtis.github.io/articles/Building-Edit-Forms.html) describes an alternative solution that don't rely on **NavigationManager**.
